forte/data/extractor/feature.py

```diff
@@ -0,0 +1,108 @@
+#  Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Any, Tuple, Union
+
+
+class Feature:
+    def __init__(self,
+                 data: List[Any],
+                 pad_id: Union[int, List],
+                 dim: int):
+        """
+        Args:
+        data (List[Any]):
+            A list of features, where each feature can be the value or another
+            list of features.
+        pad_id (int or List):
+            a single integer or a nested list of integer representing <PAD>.
```

Comment:
Why do we need nested list of `pad_id`? If you want to provide different padding for different dimension, you only need a list here, where the length is the same as the dimension.

Reply:
Here are two things:
1) If the feature is scalar value, the `pad_id` will be integer (e.g. 0). If the feature is one-hot encoding, the `pad_id` will be a vector which is represented using type `List`(e.g. [0,0,1]).
2) Padding will only happen in the inner most dimension. For example:
i) [2,7,8] -> [2,7,8,0] (1 dim feature, pad_len = 4, pad_id=0)
ii) [[1,2],[3,4,5],[9]] -> [[1,2,0],[3,4,5],[9,0,0],[0,0,0]] (2 dim feature, pad_len_dim1 = 4, pad_len_dim2=3, pad_id=0) but actually only the 2nd dim (inner most) will use the pad_id
iii) [[0,1,0],[1,0,0]] -> [[0,1,0],[1,0,0],[0,0,1]] (1 dim one-hot-encoding feature, pad_len=3, pad_id=[0,0,1])

Reply:
So
1) The `pad_id` can be a list when it is something like a one-hot vector
2) The smallest unit is determined by the shape of the data and dimension. Even if the data is a 5-dimensional list, you can specify `dim=2`, and at this time the smallest unit is a 3-dim vector.

These two points are pretty difficult to be read from the documentation and are not very intuitive.

Reply:
Yes, now our understanding is the same. I agree those are bit difficult to understand. Do you mean that you want me to add more documentation to explain this?